app/forms/provider_interface/offer_wizard.rb

```diff
@@ -38,21 +44,68 @@ def valid_for_current_step?
       valid?(current_step.to_sym)
     end
 
-    def next_step
-      index = STEPS[decision.to_sym].index(current_step.to_sym)
+    def next_step(step = current_step)
```

Comment:
> Can this be `next_step!` instead, since it changes the `current_step`?

It cant be next_step! because even though it sets the value of the next step, it doesn't save it to the data store. 

> Also, do we need the parameter here? Can we just refer to instance's current_step everywhere inside these methods?

The intention of using the name current_step is not to override the value of the instance varaible current_step as we dont know yet if the user will click next or not. This is why we want to specify the parameter when we want to calculate the step (in case any steps are missed). However this does raise a question about the choice of the variable name that we use within next step. We will rename current_step to next_step for clarity.